src/pages/api/user/register.ts

```diff
@@ -1,34 +1,67 @@
-// Next.js API route support: https://nextjs.org/docs/api-routes/introduction
-import type { NextApiRequest, NextApiResponse } from 'next'
-
-export interface RegisterRequestType {
-    email: string,
-    password: string,
-    firstName: string,
-    lastName: string,
-}
-
-export interface RegisterResponseType {
-  error: string,
-}
-
-export default async function handler(
-  req: NextApiRequest,
-  res: NextApiResponse<Response>
-) {
-  const { email, password, firstName, lastName } = req.body;
-
-  if (!email || !password || !firstName || !lastName) {
-    res.status(400);
-
-    // Maybe go?
-    throw new Error("Please add all fields");
-  }
-
-  //! Check if email already exists
-
-  //! Hash password
-
-  //! Return token
-
-}
\ No newline at end of file
+import type { NextApiRequest, NextApiResponse } from "next";
+import { prisma } from '../../../lib/prisma'
+
+// API Inputs.
+export interface RegisterRequestType {
+    firstName: string,
+    lastName: string,
+    email: string,
+    password: string,
+}
+
+// API Outputs.
+export interface RegisterResponseType {
+    error: string,
+}
+
+export default async function handler (
+    req: NextApiRequest,
+    res: NextApiResponse<RegisterResponseType>
+) {
+    const { firstName, lastName, email, password } = req.body;
+
+    // Error: Email already exists.
+    const user = await prisma.user.findFirst({
+        where: {
+            email: email,
+        },
+    })
+    
+    if(user) {
+        res.status(400);
+        
+        throw new Error("User already exists.")
+    }
+
+    // Error: Not all fields are filled out.
+    if(!firstName || !lastName || !email || !password) {
+        res.status(400);
+
+        throw new Error("Please add all fields.");
+    }
+
+    // Error: Name values are not strings.
+    if((typeof firstName !== 'string') || (typeof lastName !== 'string')) {
+        res.status(400);
+
+        throw new Error("Please input a proper name.");
+    }
+
+    // Error: Not a valid email.
+    if((!email.includes('@')) || (!email.includes('.'))) {
+        res.status(400);
+
+        throw new Error("Please input a valid email.");
+    }
+
+    // If all checks are passed.
+    
+    // Hash Password.
+    
+    // Save New User.
+    //const userData = req.body;
+    const userData = {firstName:firstName, lastName:lastName, email:email, password:password};
+    const savedUser = await prisma.user.create({data:userData});

```

Comment:
Since the key and value are the same name, this can be condensed to `data: { firstName, lastName, email, password }` and line 63 can be deleted.

Reply:
The reason I did it this way was to set up for the password hashing later.